bitcointranscripts/main.py

```diff
@@ -98,25 +98,26 @@ def parse_post(file_path):
     front_matter, body = parse_markdown(content)
     # Extract metadata from front matter using yaml
     metadata = yaml.safe_load(front_matter)
-    url_path = file_path.replace('.md', '').replace(GLOBAL_URL_VARIABLE, '')
+    url_path = os.path.normpath(file_path.replace('.md', '').replace(GLOBAL_URL_VARIABLE, ''))
     transcript_by = metadata.get('transcript_by', "")
 
     document = {
-        'id': "bitcointranscripts" + url_path.replace("\\", "+").replace("/", "+"),
+        'id': "bitcointranscripts" + url_path.replace(os.sep, "+"),
         'title': metadata['title'],
         'body_formatted': body,
         'body': body,
         'body_type': "markdown",
         'created_at': (datetime.strptime(metadata['date'], '%Y-%m-%d').strftime('%Y-%m-%dT%H:%M:%S.000Z') if isinstance(metadata.get('date'), str) else None),
         'domain': "https://btctranscripts.com/",
-        'url': "https://btctranscripts.com" + url_path.replace("\\", "/"),
+        'url': "https://btctranscripts.com" + url_path.replace(os.sep, "/"),
         'categories': metadata.get('categories', []),
         'tags': metadata.get('tags', []),
         'media': metadata.get('media', ""),
         'authors': metadata.get('speakers', []),
         'indexed_at': datetime.now().isoformat(),
         'transcript_by': transcript_by,
-        'needs_review': "Yes" if "--needs-review" in transcript_by else "No"
+        'needs_review': "Yes" if "--needs-review" in transcript_by else "No",
+        'transcript_source': url_path.split(os.sep)[1]
```

Comment:
If the current transcript is [this one](https://btctranscripts.com/advancing-bitcoin/2022/challenges-in-implementing-coinjoin-in-hardware-wallets/), then based on my understanding, `url_path` will be equal to "advancing-bitcoin/2022/challenges-in-implementing-coinjoin-in-hardware-wallets", which means that this code will pick "2022" instead of "advancing-bitcoin"

Reply:
No, it will pick "advancing-bitcoin" only. Because we are generating `url_path` based on the `file_path` from where we have downloaded the posts, after removing `./tmp/chaincode-scrapper\bitcointranscripts\bitcointranscripts-master` part from the post.
```
url_path = os.path.normpath(file_path.replace('.md', '').replace(GLOBAL_URL_VARIABLE, ''))
```

Reply:
So what would be the value of the `url_path` variable in the example that I gave?

Reply:
The `url_path` would be `/advancing-bitcoin/2022/challenges-in-implementing-coinjoin-in-hardware-wallets`

Reply:
ok, thanks for the clarification!